.github/workflows/build.yml

```diff
@@ -0,0 +1,40 @@
+name: Build
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.20'
+
+      - name: Setup TinyGo
+        uses: acifani/setup-tinygo@v2
+        with:
+          tinygo-version: '0.28.0'
+
+      - name: Setup Spin
+        uses: fermyon/actions/spin/setup@v1
+        with:
+          version: "v2.6.0"
+
+      - name: Setup Wasmtime
+        uses: bytecodealliance/actions/wasmtime/setup@v1
+        with:
+          version: "13.0.1"
```

Comment:
hmm, do we want to use one of the latest version of wasmtime maybe?

Reply:
We need wasmtime 13 because it's the latest version that works with TinyGo 0.28. TinyGo 0.29 is when `net/http` was removed and all the `Transport` bits in the client stopped working. I'll a some note about somewhere

Reply:
Specifically `--map-dir` was removed from wasmtime in v14